packages/grafana-data/src/transformations/transformers/labelsToFields.ts

```diff
@@ -77,3 +96,38 @@ export const labelsToFieldsTransformer: SynchronousDataTransformerInfo<LabelsToF
     return result;
   },
 };
+
+function convertLabelsToRows(data: DataFrame[], keepLabels?: string[]): DataFrame[] {
+  const result: DataFrame[] = [];
+  for (const frame of data) {
+    for (const field of frame.fields) {
+      if (field.labels) {
+        const keys: string[] = [];
+        const vals: string[] = [];
+        if (keepLabels) {
+          for (const key of keepLabels) {
+            keys.push(key);
+            vals.push(field.labels[key]);
+          }
+        } else {
+          for (const [key, val] of Object.entries(field.labels)) {
+            keys.push(key);
+            vals.push(val);
+          }
+        }
+        if (vals.length) {
+          result.push({
+            ...frame,
+            name: getFieldDisplayName(field, frame, data),
+            fields: [
+              { name: 'label', type: FieldType.string, config: {}, values: new ArrayVector(keys) },
+              { name: 'value', type: FieldType.string, config: {}, values: new ArrayVector(vals) },
```

Comment:
Should we capitalize these (Label vs label) , Value vs value? not sure :shrug:  , but I think time series field names are Time and Value (ie capitalized)

Reply:
when capitalized special magic name handling takes over :(  

it ends up with the frame name in the display unless we add additional logic :shrug: